ambry-account/src/main/java/com/github/ambry/account/AccountUtils.java

```diff
@@ -70,4 +73,95 @@ static boolean hasDuplicateAccountIdOrName(Collection<Account> accounts) {
+    accountsInSecondary.forEach(account -> secondaryAccountMap.put(account.getId(), account));
+
+    Set<Account> accountsMissingInSecondary = accountsInPrimary.stream()
+        .filter(account -> secondaryAccountMap.get(account.getId()) == null)
+        .collect(Collectors.toSet());
+
+    Set<Account> accountsDifferentInSecondary = new HashSet<>(accountsInPrimary);
+    accountsDifferentInSecondary.removeAll(accountsInSecondary);
+    accountsDifferentInSecondary.removeAll(accountsMissingInSecondary);
+
+    if (!accountsMissingInSecondary.isEmpty() || !accountsDifferentInSecondary.isEmpty()) {
+      isSame = false;
+
+      StringBuilder accountsInfo = new StringBuilder();
+
+      if (!accountsMissingInSecondary.isEmpty()) {
+        accountsInfo.append("[");
+        for (Account account : accountsMissingInSecondary) {
+          accountsInfo.append(AccountCollectionSerde.accountToJsonNoContainers(account).toString()).append(",");
+        }
+        accountsInfo.append("]");
+        logger.warn("Accounts found in primary and absent in secondary = {}", accountsInfo.toString());
+      }
+
+      if (!accountsDifferentInSecondary.isEmpty()) {
+        accountsInfo.setLength(0);
+        accountsInfo.append("[");
+        for (Account account : accountsDifferentInSecondary) {
+          accountsInfo.append("{Account = ")
+              .append(account.toString())
+              .append(", primary = ")
+              .append(AccountCollectionSerde.accountToJsonNoContainers(account).toString())
+              .append(", secondary = ")
+              .append(AccountCollectionSerde.accountToJsonNoContainers(secondaryAccountMap.get(account.getId()))
+                  .toString());
+
+          Set<Container> containersMissingInSecondary = account.getAllContainers()
+              .stream()
+              .filter(
+                  container -> secondaryAccountMap.get(account.getId()).getContainerByName(container.getName()) == null)
+              .collect(Collectors.toSet());
+
+          Set<Container> containersDifferentInSecondary = new HashSet<>(account.getAllContainers());
+          containersDifferentInSecondary.removeAll(secondaryAccountMap.get(account.getId()).getAllContainers());
+          containersDifferentInSecondary.removeAll(containersMissingInSecondary);
+
+          if (!containersMissingInSecondary.isEmpty()) {
+            accountsInfo.append(", Containers missing in secondary: [");
+            for (Container container : containersMissingInSecondary) {
+              accountsInfo.append(container.toJson().toString()).append(",");
+            }
+            accountsInfo.append("]");
+          }
+
+          if (!containersDifferentInSecondary.isEmpty()) {
+            accountsInfo.append(", Containers different in secondary: [");
+            for (Container container : containersDifferentInSecondary) {
+              accountsInfo.append("{container = ")
+                  .append(container.toString())
```

Comment:
`container.toString()` doesn't have container name, is this ok? (I guess probably ok because I see you have `container.toJson().toString()` in following lines)